src/solvers/refinement/bv_refinement.h

```diff
@@ -33,6 +31,12 @@ class bv_refinementt:public bv_pointerst
     /// Enable arithmetic refinement
     bool refine_arithmetic=true;
   };
+public:
+  struct infot:public configt
+  {
+    const namespacet *ns=nullptr;
```

Comment:
Making this hold a reference would be better instead of reintroducing pointers...

Reply:
This is the solution to the problem of functions requiring too many arguments. Either we'd use this (essence pattern) or the builder pattern. Both of which use null pointers to hold the default value for a reference (but builder is more verbose and not composable).

The only solution that allows usage of both arguments and references at the same time, are C++20 named arguments.